pkg/standalone-utils/contracts/relayer/VaultActions.sol

```diff
@@ -236,52 +236,77 @@ abstract contract VaultActions is IBaseRelayerLibrary {
-            return _doWeightedExactBptInForOneTokenOutReplacements(userData);
+            return _doWeightedExactBptInForOneTokenOutReplacements(poolId, userData);
         } else if (kind == WeightedPoolUserData.ExitKind.EXACT_BPT_IN_FOR_TOKENS_OUT) {
-            return _doWeightedExactBptInForTokensOutReplacements(userData);
+            return _doWeightedExactBptInForTokensOutReplacements(poolId, userData);
         } else {
             // All other exit kinds are 'given out' (i.e the parameter is a token amount),
             // so we don't do replacements for those.
             return userData;
         }
     }
 
-    function _doWeightedExactBptInForOneTokenOutReplacements(bytes memory userData) private returns (bytes memory) {
+    function _doWeightedExactBptInForOneTokenOutReplacements(bytes32 poolId, bytes memory userData)
+        private
+        returns (bytes memory)
+    {
         (uint256 bptAmountIn, uint256 tokenIndex) = WeightedPoolUserData.exactBptInForTokenOut(userData);
 
         if (_isChainedReference(bptAmountIn)) {
             bptAmountIn = _getChainedReferenceValue(bptAmountIn);
             return abi.encode(WeightedPoolUserData.ExitKind.EXACT_BPT_IN_FOR_ONE_TOKEN_OUT, bptAmountIn, tokenIndex);
+        } else if (_useRelayerBalance(bptAmountIn)) {
+            // TODO(@jubeira): require sender == address(this)?
+            bptAmountIn = IERC20(VaultHelpers.toPoolAddress(poolId)).balanceOf(address(this));
+            return abi.encode(WeightedPoolUserData.ExitKind.EXACT_BPT_IN_FOR_ONE_TOKEN_OUT, bptAmountIn, tokenIndex);
         } else {
             // Save gas by only re-encoding the data if we actually performed a replacement
             return userData;
         }
     }
 
-    function _doWeightedExactBptInForTokensOutReplacements(bytes memory userData) private returns (bytes memory) {
+    function _doWeightedExactBptInForTokensOutReplacements(bytes32 poolId, bytes memory userData)
+        private
+        returns (bytes memory)
+    {
         uint256 bptAmountIn = WeightedPoolUserData.exactBptInForTokensOut(userData);
 
         if (_isChainedReference(bptAmountIn)) {
             bptAmountIn = _getChainedReferenceValue(bptAmountIn);
             return abi.encode(WeightedPoolUserData.ExitKind.EXACT_BPT_IN_FOR_TOKENS_OUT, bptAmountIn);
+        } else if (_useRelayerBalance(bptAmountIn)) {
+            // TODO(@jubeira): require sender == address(this)?
+            bptAmountIn = IERC20(VaultHelpers.toPoolAddress(poolId)).balanceOf(address(this));
+            return abi.encode(WeightedPoolUserData.ExitKind.EXACT_BPT_IN_FOR_TOKENS_OUT, bptAmountIn);
         } else {
             // Save gas by only re-encoding the data if we actually performed a replacement
             return userData;
         }
     }
+
+    function _useRelayerBalance(uint256 bptAmountIn) internal pure returns (bool) {
+        // Look for 0xa11 sentinel value to use all of relayer's balance.
```

Comment:
Lol does this stand for 'all'? :joy:

Reply:
if `0xba1` stands for BAL, why not? :upside_down_face: